includes/validation/class-amp-validated-url-post-type.php

```diff
@@ -735,51 +735,48 @@ public static function store_validation_errors( $validation_errors, $url, $args
 			$term_data = AMP_Validation_Error_Taxonomy::prepare_validation_error_taxonomy_term( $data );
 			$term_slug = $term_data['slug'];
 
-			if ( ! isset( $terms[ $term_slug ] ) ) {
-
-				// Not using WP_Term_Query since more likely individual terms are cached and wp_insert_term() will itself look at this cache anyway.
-				$term = AMP_Validation_Error_Taxonomy::get_term( $term_slug );
-				if ( ! ( $term instanceof WP_Term ) ) {
-					/*
-					 * The default term_group is 0 so that is AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_REJECTED_STATUS.
-					 * If sanitization auto-acceptance is enabled, then the term_group will be updated below.
-					 */
-					$r = wp_insert_term( $term_slug, AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG, wp_slash( $term_data ) );
-					if ( is_wp_error( $r ) ) {
-						continue;
-					}
-					$term_id = $r['term_id'];
-					update_term_meta( $term_id, 'created_date_gmt', current_time( 'mysql', true ) );
-
-					/*
-					 * When sanitization is forced by filter, make sure the term is created with the filtered status.
-					 * For some reason, the wp_insert_term() function doesn't work with the term_group being passed in.
-					 */
-					$sanitization = AMP_Validation_Error_Taxonomy::get_validation_error_sanitization( $data );
-					if ( 'with_filter' === $sanitization['forced'] ) {
-						$term_data['term_group'] = $sanitization['status'];
-						wp_update_term(
-							$term_id,
-							AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG,
-							[
-								'term_group' => $sanitization['status'],
-							]
-						);
-					} elseif ( AMP_Validation_Manager::is_sanitization_auto_accepted( $data ) ) {
-						$term_data['term_group'] = AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_ACCEPTED_STATUS;
-						wp_update_term(
-							$term_id,
-							AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG,
-							[
-								'term_group' => $term_data['term_group'],
-							]
-						);
-					}
-
-					$term = get_term( $term_id );
+			// Not using WP_Term_Query since more likely individual terms are cached and wp_insert_term() will itself look at this cache anyway.
```

Comment:
I've only removed the unneeded if statement here, since `$terms` will always be an empty array.